pkg/diskmaker/discovery/monitor.go

```diff
@@ -0,0 +1,123 @@
+package discovery
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/klog"
+)
+
+// Monitors udev for block device changes, and collapses these events such that
+// only one event is emitted per period in order to deal with flapping.
+func udevBlockMonitor(c chan string, period time.Duration) {
+	defer close(c)
+	var udevFilter []string
+
+	// return any add or remove events, but none that match device mapper
+	// events. string matching is case-insensitive
+	events := make(chan string)
+
+	// regex to ignore the list of devices where udev events should not
+	// trigger continous discovery
+	discoverUdev := "(?i)dm-[0-9]+,(?i)rbd[0-9]+,(?i)nbd[0-9]+"
```

Comment:
Lets move this string to a constant. But also, why does this variable have to be initialized at all, if we are only creating it for the purpose of splitting into list of filters? Shouldn't we just initialize the filter list directly?